sdk/communication/azure-communication-callautomation/azure/communication/callautomation/_call_automation_client.py

```diff
@@ -426,31 +498,34 @@ def start_recording(
         :rtype: ~azure.communication.callautomation.RecordingProperties
         :raises ~azure.core.exceptions.HttpResponseError:
         """
-        channel_affinity_internal = []
 
-        if channel_affinity:
-            for channel in channel_affinity:
-                channel_affinity_internal.append(channel._to_generated(# pylint:disable=protected-access
-                    ))
+    @distributed_trace
+    def start_recording(
+        self,
+        *args: Union['ServerCallLocator', 'GroupCallLocator'],
```

Comment:
Should this accept `str` as well?

Reply:
The `str` values are keyword-only - so `args` should only be the deprecated models.